ui/src/api.ts

```diff
@@ -25,14 +25,25 @@ export async function getAllPapers() {
 }
 
 export async function papers(s2Ids: string[]) {
-  const data = await doGet(
-    axios.get("/api/v0/papers", {
-      params: {
-        id: s2Ids.join(",")
-      }
-    })
-  );
-  return (data || []) as Paper[];
+  const PAPER_REQUEST_BATCH_SIZE = 50;
+  var paperChunk = [];
+  var promises = [];
+
+  for (var i = 0; i < s2Ids.length; i += PAPER_REQUEST_BATCH_SIZE) {
```

Comment:
Can you change the variable declarations for the loop variable (here and on Line 36) to:

```suggestion
  for (let i = 0; i < s2Ids.length; i += PAPER_REQUEST_BATCH_SIZE) {
```

to make it clear that the scope of `i` is limited to the current block? (instead of the default global scope of `var` declarations? Check out https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let)